exercises/binary-search-tree/binary_search_tree_test.go

```diff
@@ -101,38 +113,62 @@ func TestComplexTree(t *testing.T) {
 		t.Errorf("bst.data: %d, want %d.", actual, expected)
 	}
 
-	actual = bst.left.data
 	expected = 2
+	if bst.left == nil {
+		t.Errorf("bst.left: nil, want %d.", expected)
+		return
```

Comment:
I have a situation to share, but I am not sure if it is a problem worth solving. Here goes.

Let's say in response to this test, I make a `bst` whose `data` is 4 and both `left` and `right` are nil.

The old test would panic, this is obviously bad.
The new test doesn't panic, tells me `bst.left` is `nil`, much better!

The question is: can/should the test also go on to tell me that in addition to `bst.left` being `nil`, `bst.right` is also nil?

The interesting thing is that if previously I had created a tree whose node values were all 0, because there were no `return` nor `Fatalf`, the test would have gone through and told me **everything wrong with the tree**. Now, if we run into a `nil`, we stop looking for any other problems with the tree.

Feel free to decide that the behaviour in this PR is fine, since the proposed addition of finding everything wrong with the tree would probably involve putting more things in conditionals and the indenting will make the change larger.